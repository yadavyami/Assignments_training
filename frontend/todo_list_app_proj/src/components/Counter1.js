import React, { useState } from 'react';

const Counter1 = () => {
  const [count, setCount] = useState(0);

  const increment = () => {
    setCount(count + 1);
  };

  return (
    <div>
      <p>Count: {count}</p>
      <button onClick={increment} test-id="button">Increment</button>
    </div>
  );
};

export default Counter1;