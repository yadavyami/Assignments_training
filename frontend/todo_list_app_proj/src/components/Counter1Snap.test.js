import React from 'react'
import renderer from 'react-test-renderer'
import Counter1 from './Counter1'


test("Counter1 snapshot with different count", () => {
    const component = renderer.create(<Counter1 />)

    // initial rendering
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    
    // simulate a user clicking the button
    tree.children[1].props.onClick();
    
    // re-rendering after the click event
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})