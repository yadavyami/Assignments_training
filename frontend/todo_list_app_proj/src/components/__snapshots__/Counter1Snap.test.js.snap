// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter1 snapshot with different count 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <button
    onClick={[Function]}
    test-id="button"
  >
    Increment
  </button>
</div>
`;

exports[`Counter1 snapshot with different count 2`] = `
<div>
  <p>
    Count: 
    1
  </p>
  <button
    onClick={[Function]}
    test-id="button"
  >
    Increment
  </button>
</div>
`;
